iib/workers/tasks/build_regenerate_bundle.py

```diff
@@ -367,14 +379,55 @@ def _resolve_image_pull_specs(bundle_metadata, labels, pinned_by_iib, registry_r
             log.debug('%s will be replaced with %s', pullspec, new_pullspec.to_str())
             replacement_pullspecs[pullspec] = new_pullspec
 
+    _replace_csv_pullspecs(bundle_metadata, replacement_pullspecs)
+
+
+def _replace_image_name_from_labels(bundle_metadata, replacement_template):
+    """
+    Replace repo/image-name in the CSV pull specs with values from their labels.
+
+    :param dict bundle_metadata: the dictionary of CSV's and relatedImages pull specifications
+    :param str replacement_template: the template specifying which label values to use for
+        replacement
+    """
+    replacement_pullspecs = {}
+
+    for pullspec in bundle_metadata['found_pullspecs']:
+        new_pullspec = ImageName.parse(pullspec.to_str())
+        pullspec_labels = get_image_labels(pullspec.to_str())
+        try:
+            modified_namespace_repo = replacement_template.format(**pullspec_labels)
+        except KeyError:
+            raise IIBError(
+                f'Pull spec {pullspec.to_str()} is missing one or more label(s)'
+                f' required in the image_name_from_labels {replacement_template}.'
+                f' Available labels: {", ".join(list(pullspec_labels.keys()))}'
+            )
+        new_pullspec.namespace, new_pullspec.repo = modified_namespace_repo.split('/', 1)
```

Comment:
What if the `modified_namespace_repo` does not contain `/` ?

Reply:
Good point

Reply:
Note from meeting:

* If there are two items after split, then the first is namespace, the second is repo. (as is in code right now)
* If there is only one item after split, then namespaces is `None`, and the only item is repo.

Reply:
Done